apps/decoder_csv/lib/decoder/csv.ex

```diff
@@ -25,25 +25,25 @@ defmodule Decoder.Csv do
   defimpl Decoder, for: __MODULE__ do
     def lines_or_bytes(_t), do: :line
 
-    def decode(t, stream) do
-      stream
-      |> Stream.transform(%{skip: t.skip_first_line}, fn chunk, %{skip: skip} = acc ->
-        parsed_chunk = parse_chunk(chunk, t.headers, skip)
-        {[parsed_chunk], %{acc | skip: false}}
-      end)
+    def decode(t, messages) do
+      do_decode(messages, t.skip_first_line and !already_skipped?())
```

Comment:
I might pull t.skip_first_line and !already_skipped?() in a super small function which returned the same result and name it should_skip().  I'm not strongly opinionated about this as it seems like possibly too much, but i dont like a method being called with a `and` clause as part of one of the parameters